File: core/src/main/java/com/uiptv/service/remotesync/VerificationCodeGenerator.java
package com.uiptv.service.remotesync;

import java.security.SecureRandom;

public final class VerificationCodeGenerator {
    private static final SecureRandom RANDOM = new SecureRandom();

    private VerificationCodeGenerator() {
    }

    public static String createFourDigitCode() {
        return String.format("%04d", RANDOM.nextInt(10_000));
    }
}

File: server/src/main/kotlin/com/uiptv/service/remotesync/VerificationCodeGenerator.kt
package com.uiptv.service.remotesync

import java.security.SecureRandom

object VerificationCodeGenerator {
    private val random = SecureRandom()

    @JvmStatic
    fun createFourDigitCode(): String = "%04d".format(random.nextInt(10_000))
}
